awscli/customizations/s3/fileinfo.py

```diff
@@ -263,9 +266,10 @@ def _handle_upload(self, body):
         }
         self._handle_object_params(params)
         response_data = self.client.put_object(**params)
-        etag = response_data['ETag'][1:-1]
-        body.seek(0)
-        check_etag(etag, body)
+        if response_data['ServerSideEncryption'] != 'aws:kms':
```

Comment:
Much like #1026, I had to disable the etag checking because of the difference in MD5ness between encrypted and non-encrypted.